sem-install

```diff
@@ -0,0 +1,81 @@
+#!/bin/bash
+HOST="http://packages.semaphoreci.com"
+
+install::check_precompiled() {
+  response=$(curl --write-out %{http_code} --head --silent --output /dev/null $HOST/${language}/${language_version}.tar.gz)
+  echo $response
+}
+
+install::dl_precompiled() {
+  curl --silent $HOST/${language}/${language_version}.tar.gz --output ${language_version}.tar.gz
+}
+
+install::install_ruby() {
+  response=$(install::check_precompiled)
+  if [[ $response -ne 200 ]] ; then
+    rbenv install -s ${language_version}
+  else
+    install::dl_precompiled
+    tar -zxf ${language_version}.tar.gz
+    mv ${language_version} ~/.rbenv/versions/
+  fi
+  rbenv rehash
+  return $?
+}
+
+install::install_php() {
+  resp=$(install::check_precompiled)
+  if [[ $resp -ne 200 ]] ; then
+    phpbrew install -j 4 ${language_version} +default +dbs +openssl -- --with-curl=/usr/local
+    echo "date.timezone = UTC" >> "~/.phpbrew/php/php-${language_version}/etc/php.ini"
+    phpbrew ext install imap -- --with-kerberos --with-imap-ssl
+    phpbrew ext install gd -- --enable-gd-native-ttf --with-freetype-dir=/usr/lib/x86_64-linux-gnu
+  else
+    echo $(install::dl_precompiled)
+    tar -zxf ${language_version}.tar.gz 
+    mv php-${language_version} ~/.phpbrew/php/
+    if [ ! -d "~/.phpbrew/build/php-{$language_version}" ]; then
+	phpbrew download ${language_version}
+	tar -xjf ~/.phpbrew/distfiles/php-${language_version}.tar.bz2
+	mv php-${language_version} ~/.phpbrew/build/
+    fi
+  fi
+  return $?
+}
+
+install::usage() {
+  echo -e "
+Usage:
+  sem-install language version"
+}
+
+install::err() {
+  echo -e "\n! [$(version::date)]: $@" >&2
+
+  return 1
+}
+
+sem-install() { 
+  if ! [[ $# -eq 2 ]]; then
+    install::usage
+    install::err "Unsupported number of arguments provided!"
+
+    return $?
+  fi
+
+  language=$1
+  language_version=$2
+
+  case $language in
+    php)
+      install::install_php 
+      ;;
+    ruby)
+      install::install_ruby 
+      ;;
+    *)
+      install:err "Language '${language}' install not supported yet."
```

Comment:
`Installation of '${language}' is not yet supported` or `Unsupported language '${language}'`.

Reply:
Done